src/dataDisplay/tableHeaderCell.jsx

```diff
@@ -1,32 +1,363 @@
-'use strict';
-
-import React, { Component } from 'react';
-import _ from 'lodash';
 import ClassNames from 'classnames';
+import _ from 'lodash';
 import PropTypes from 'prop-types';
-import Utils from '../utils/utils.js';
+import React from 'react';
+import { Resizable } from 'react-resizable';
+import Utils from '../utils/utils';
+import Icon from './icon';
+import { values as breakPoints } from '../styles/breakpointsConstants';
+
+const columnNumberEnums = ['auto', 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const propTypes = {
+    adjacentStickyColumnsTotalWidth: PropTypes.number,
+    children: PropTypes.node,
+    className: PropTypes.string,
+    collapsing: PropTypes.bool,
+    desktop: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.oneOf(columnNumberEnums),
+    ]),
+    desktopLarge: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.oneOf(columnNumberEnums),
+    ]),
+    forceTableUpdate: PropTypes.func,
+    id: PropTypes.string,
+    isResizable: PropTypes.bool,
+    laptop: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.oneOf(columnNumberEnums),
+    ]),
+    mobile: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.oneOf(columnNumberEnums),
+    ]),
+    mobileLarge: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.oneOf(columnNumberEnums),
+    ]),
+    mobileMedium: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.oneOf(columnNumberEnums),
+    ]),
+    onClick: PropTypes.func,
+    resizableColumnWidthPercentage: PropTypes.number,
+    setResizableCellsWordWrapping: PropTypes.func,
+    shouldResetWhiteSpaceStyle: PropTypes.bool,
+    sticky: PropTypes.bool,
+    stickyColumnCount: PropTypes.number,
+    stickyTableContainerWidth: PropTypes.number,
+    style: PropTypes.object,
+    tablet: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.oneOf(columnNumberEnums),
+    ]),
+    textAlign: PropTypes.oneOf(['center', 'left', 'right']),
+    width: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.oneOf(columnNumberEnums),
+    ]),
+};
+
+const defaultProps = {
+    adjacentStickyColumnsTotalWidth: 0,
+    forceTableUpdate: undefined,
+    isResizable: false,
+    resizableColumnWidthPercentage: undefined,
+    stickyColumnCount: undefined,
+    setResizableCellsWordWrapping: undefined,
+    shouldResetWhiteSpaceStyle: true,
+    stickyTableContainerWidth: 0,
+};
+
+const HEADER_CELL_WIDTH_OFFSET = 10;
+const DEBOUNCE_WAIT_TIME = 50;
+export const MAX_WIDTH_PERCENTAGE_DESKTOP = 0.5;
+export const MAX_WIDTH_PERCENTAGE_MOBILE = 0.8;
```

Comment:
Can you please move these exports to a tableConstants.js file.

Reply:
done!